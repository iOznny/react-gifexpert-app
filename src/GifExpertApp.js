import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['One Punch']);

    // Renderizar el componente.
    return (
        <Fragment>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={ setCategories } />

            <hr />            

            <ol>
                {
                    categories.map(category => (                        
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </Fragment>
    )
}