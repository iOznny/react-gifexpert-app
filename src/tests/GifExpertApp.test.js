import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en GifExpertApp', () => {
    test('Show successful', () => {
       const wrapper = shallow(<GifExpertApp />);
       expect(wrapper).toMatchSnapshot();
    });
    
    test('Show list of categories', () => {
        const categories = ['One Punch', 'Space X'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });    
});
