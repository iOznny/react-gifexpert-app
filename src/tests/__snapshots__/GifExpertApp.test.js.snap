// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Show list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Space X"
      key="Space X"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Show successful 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
